x/wasm/internal/types/msg.go

```diff
@@ -26,9 +36,26 @@ func (msg MsgStoreCode) ValidateBasic() sdk.Error {
 	if len(msg.WASMByteCode) == 0 {
 		return sdk.ErrInternal("empty wasm code")
 	}
+
 	if len(msg.WASMByteCode) > MaxWasmSize {
 		return sdk.ErrInternal("wasm code too large")
 	}
+
+	if msg.Source != "" {
+		fmt.Println("source: ", msg.Source)
+
+		if !govalidator.IsURL(msg.Source) {
```

Comment:
I'm not strongly against adding more dependencies, but I like to see the utility.
Bringing in govalidator for one line seems too much.

You can use: https://golang.org/pkg/net/url/#Parse from stdlib, which should have the same functionality

Reply:
We used `net/url.Parse()` previously but the method has a few weaknesses. If you change the url value to `somerandomstring#43243@` also `net/url.Parse()` function will still pass the broken URL as `valid`. This is NOT the right way to validate URL.

Reply:
May be we can do the following.
- check if url is reachable
- check if the url is absolute (we shouldn't encourage relative urls for source anyway)

Reply:
Added these changes and updated genesis tests to cover these cases.